test/torch/tensors/test_autograd.py

```diff
@@ -541,3 +541,73 @@ def test_get_float_prec_on_autograd_tensor(workers):
     x2 = x.fix_precision()
     x2 = x2.share(bob, alice, crypto_provider=james, requires_grad=True)
     assert (x2.get().float_precision() == x).all()
+
+
+def test_serialize_deserialize_autograd_tensor(workers):
+    # let's try to send an autogradTensor to a remote location and get it back
+    alice = workers["alice"]
+    random_tensor = torch.randn(5, 3)
+    remote_tensor = random_tensor.send(alice, local_autograd=True)
+    local_tensor = remote_tensor.get()
+    # check if the tensor sent is equal to the tensor got from the remote version
+    assert torch.all(torch.eq(local_tensor, random_tensor))
+
+
+def test_train_remote_autograd_tensor(workers):
+    # Training procedure to train an input model, be it remote or local
+
+    def train(model_input, data_input, target_input, remote=False):
+        opt = optim.SGD(params=model_input.parameters(), lr=0.1)
+        loss_previous = 99999999999  # just a very big number
+        for iter in range(10):
+            # 1) erase previous gradients (if they exist)
+            opt.zero_grad()
+            # 2) make a prediction
+            predictions = model_input(data_input)
+            # 3) calculate how much we missed
+            loss = ((predictions - target_input) ** 2).sum()
+            # check for monotonic decrease of the loss
+
+            if remote == True:
+                # Remote loss monotonic decrease
+                loss_val_local = loss.copy().get().item()
+                assert loss_val_local < loss_previous
+                loss_previous = loss_val_local
+
+            else:
+                # Local loss monotonic decrease
+                loss_val_local = loss.item()
+                assert loss_val_local < loss_previous
+                loss_previous = loss_val_local
+
+            # 4) Figure out which weights caused us to miss
+            loss.backward()
+            # 5) change those weights
+            opt.step()
+        return (loss, model_input)
+
+    worker = workers["alice"]
+
+    # Some Toy Data
+    data_local = torch.tensor([[0, 0], [0, 1], [1, 0], [1, 1.0]])
+    target_local = torch.tensor([[0], [0], [1], [1.0]])
+
+    # Toy local model
+    model_local = nn.Linear(2, 1)
+
+    # Local training
+    loss_local, model_local_trained = train(model_local, data_local, target_local, remote=False)
+
+    # Remote training, setting autograd for the data and the targets
+    data_remote = data_local.send(worker, local_autograd=True)
+    target_remote = target_local.send(worker, local_autograd=True)
+    model_remote = model_local.send(worker, local_autograd=True)
```

Comment:
So there is a bug here: Autograd should be created because model parameters have require_grad to True but it's not currently working :/